Ginger/GingerCoreCommon/Repository/ApplicationModelLib/ApplicationModelBase.cs

```diff
@@ -138,34 +138,59 @@ public bool SupportSimulation
+                {
+                    if (mi.Name == "mName" || mi.Name == "Name" || mi.Name == "ItemName" || mi.Name == "RelativeFilePath" || mi.Name == "FilePath" || mi.Name == "ObjFileExt" || mi.Name == "ObjFolderName" || mi.Name == "ItemNameField") continue;
 
-                PropertyInfo PI = this.GetType().GetProperty(mi.Name);
-                dynamic value = null;
-                if (mi.MemberType == MemberTypes.Property)
-                    value = PI.GetValue(this);
-                else if (mi.MemberType == MemberTypes.Field)
-                    value = this.GetType().GetField(mi.Name).GetValue(this);
+                    PropertyInfo PI = obj.GetType().GetProperty(mi.Name);
+                    dynamic value = null;
+                    if (mi.MemberType == MemberTypes.Property)
+                        value = PI.GetValue(obj);
+                    else if (mi.MemberType == MemberTypes.Field)
+                        value = obj.GetType().GetField(mi.Name).GetValue(obj);
 
-                if (value != null && value is string)
-                {
-                    string valueString = (string)PI.GetValue(this);
-                    foreach (AppModelParameter AMP in AppModelParameters)
-                    {
-                        valueString = valueString.Replace(AMP.PlaceHolder, AMP.ExecutionValue);
-                        PI.SetValue(this, valueString);
-                    }
-                    foreach (GlobalAppModelParameter GAMP in GlobalAppModelParameters)
+                    if (value != null)
                     {
-                        valueString = valueString.Replace(GAMP.PlaceHolder, GAMP.ExecutionValue);
-                        PI.SetValue(this, valueString);
+                        if (value is string)
+                        {
+                            string valueString = (string)PI.GetValue(obj);
+                            foreach (AppModelParameter AMP in AppModelParameters)
+                            {
+                                valueString = valueString.Replace(AMP.PlaceHolder, AMP.ExecutionValue);
+                                PI.SetValue(obj, valueString);
+                            }
+                            foreach (GlobalAppModelParameter GAMP in GlobalAppModelParameters)
+                            {
+                                valueString = valueString.Replace(GAMP.PlaceHolder, GAMP.ExecutionValue);
+                                PI.SetValue(obj, valueString);
+                            }
+                        }
+                        else if(value is IObservableList && GetModelListsToConfigsWithExecutionData().Contains(mi.Name))
+                        {
+                            foreach (object o in value)
+                            {
+                                SetModelConfigsWithExecutionData(o);
+                            }
+                        }
                     }
                 }
+                catch (Exception ex)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Handle the exception or explain in a comment why it can be ignored.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5576541)